llvm/lib/Target/RISCV/RISCVVLOptimizer.cpp

```diff
@@ -1371,18 +1369,8 @@ bool RISCVVLOptimizer::runOnMachineFunction(MachineFunction &MF) {
       if (!tryReduceVL(MI))
         continue;
       MadeChange = true;
-      PushOperands(MI, /*IgnoreSameBlock*/ true);
     }
   }
 
-  while (!Worklist.empty()) {
-    assert(MadeChange);
-    MachineInstr &MI = *Worklist.pop_back_val();
-    assert(isCandidate(MI));
```

Comment:
You are removing the assert that triggered the cross BB failure. Can we add back the VL=1 code above?